File: Examples/CSharp/ProgrammersGuide/Working-Shapes/Working-with-Shapes-Gluing/GetGluedConnectors.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSharp.ProgrammersGuide.Working_Shapes.Working_with_Shapes_Gluing
{
    public class GetGluedConnectors
    {
        public static void Run()
        {
            //ExStart:GetGluedConnectors
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // call a Diagram class constructor to load the VSD diagram
            Diagram diagram = new Diagram(dataDir + "RetrieveShapeInfo.vsd");
            // get shape by an ID
            Shape shape = diagram.Pages[0].Shapes.GetShape(90);
            // get all glued 1D shapes
            long[] gluedShapeIds = shape.GluedShapes(GluedShapesFlags.GluedShapesAll1D, null, null);

            // display shape ID and name
            foreach (long id in gluedShapeIds)
            {
                shape = diagram.Pages[0].Shapes.GetShape(id);
                Console.WriteLine("ID: " + shape.ID + "\t\t Name: " + shape.Name);
            }
            //ExEnd:GetGluedConnectors
        }
    }
}

File: Examples/VisualBasic/Working-Shapes/Working-with-Shapes-Gluing/GetGluedConnectors.vb

Imports Aspose.Diagram
Imports System

Public Class GetGluedConnectors
    Public Shared Sub Run()
        ' ExStart:GetGluedConnectors
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_Shapes()

        ' Call a Diagram class constructor to load the VSD diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("RetrieveShapeInfo.vsd"))
        ' Get shape by an ID
        Dim shape As Shape = diagram.Pages(0).Shapes.GetShape(90)
        ' Get all glued 1D shapes
        Dim gluedShapeIds As Long() = shape.GluedShapes(GluedShapesFlags.GluedShapesAll1D, Nothing, Nothing)

        ' Display shape ID and name
        For Each id As Long In gluedShapeIds
            shape = diagram.Pages(0).Shapes.GetShape(id)
            Console.WriteLine("ID: " & shape.ID & "\t\t Name: " & shape.Name)
        Next
        ' ExEnd:GetGluedConnectors
    End Sub
End Class
